### ISSUE TITLE
Map marshaling

### ISSUE THREAD

**Issue by XanthusL** [ugorji/go#210]
```go
        m := make(map[int32]struct{})
        m[1] = struct{}{}
        m[2] = struct{}{}
        m[3] = struct{}{}
        m[4] = struct{}{}
        m[5] = struct{}{}
        m[-1] = struct{}{}
```
The result of standard library is 
```json
{"1":{},"2":{},"3":{},"4":{},"5":{},"-1":{}}
```
But result of `go/codec` is
```
{3:{},4:{},5:{},-1:{},1:{},2:{}}
```
It's not standard json format. I have trouble with negative int keys. `-` causes an exception while parsing the json string

**Comment by ugorji** [comment]
We purposefully allowed the encoding of maps where the output is not valid json. Our vision is that any format should be adequately represented in json-like structure, meaning we relaxed the requirement of string keys, and consequently allowed most map[key]value types to be encoded/decoded, regardless of the key type.

encoding/json formerly only allowed maps with string keys, but now allow some numeric keys also.

However, enforcing strict json was always the vision, where we provided options to allow something restrict themselves to strict json. We did this well on the decoding side (via the MapType option), but not on the encoding side.

We will now add a MapStringAsKey option to jsonHandle. Once set, we will output every key using its string representation.

That should take care of your request. i.e.

use: h := &codec.JsonHandle{MapKeyAsString: true}

My next push should have the fix.
